### DISCUSSION TITLE
About Upsampling

### DISCUSSION THREAD

**Discussion by DoTulip** [google/flax#1670] [Q&A] [answered]
Hello Flax Team!
    I am trying to use FLAX for my own work. When building a neural network, I need to use upsampling (just like torch.nn.UpsamplingBilinear2d or tf.keras.layers.UpSampling2D). But I found that FLAX does not provide the corresponding function, how can I achieve upsampling?  Thank you in advance! @jheek

**Comment by matthias-wright** [comment, marked as answer]
Hi @DoTulip, you can use [`jax.image.resize`](https://jax.readthedocs.io/en/latest/_autosummary/jax.image.resize.html#jax.image.resize) for this:

```python
import jax

rng = jax.random.PRNGKey(0)

B, H, W, C = 2, 5, 5, 3
factor = 2

x = jax.random.normal(rng, (B, H, W, C))
print(x.shape)

out = jax.image.resize(x, shape=(B, H * factor, W * factor, C), method='bilinear')
print(out.shape)

```

**Reply by DoTulip** [reply]
That really helped me a lot! Thank you very much!
